### DISCUSSION TITLE
Does the DebugAdapterExecutable interface terminate the debugger process on disconnect?

### DISCUSSION THREAD

**Discussion by andrew4x** [microsoft/vscode-discussions#427] [Extension Development QnA] [answered]
Hello,

I'm working on custom programming language debugger development. The debugger is written in Java with LSP4J, and supports two communication modes - TCP connection and standard I/O.

Now, I have the following situation:
- in TCP mode everything works fine (it uses DebugAdapterServer interface), the debugger can safely stop itself on disconnect request;
- in standard I/O mode the debugger receives disconnect request, then suddenly the debugger process is being terminated (it cannot reach the end of the program), the client receives non-zero exit code.

The question is, how the VS Code synchronizes with the debugger process termination? What is expected behavior for that case?

**Comment by isidorn** [comment, marked as answer]
These docs should help https://code.visualstudio.com/blogs/2018/08/07/debug-adapter-protocol-website/
